robottelo/performance/graph.py

```diff
@@ -0,0 +1,121 @@
+"""Test utilities for generating charts for both Candlepin and Pulp tests"""
+import pygal
+
+
+def generate_bar_chart_stat(stat_dict, head, filename, lengend):
+    """Generate Bar chart for stat on concurrent subscription
+
+    :param dict stat_dict: The dictionary containing min/median/max/std
+    :param str head: Titile of charts
+    :param str filename: The name of output svg chart
+    :param str lengend: The lengend of generated bar charts
+
+    """
+    bar_chart = pygal.Bar()
+    bar_chart.title = head
+    bar_chart.x_labels = ('min', 'mean', 'max', 'std')
+    bar_chart.x_title = 'Statistics'
+    bar_chart.y_title = 'Time (s)'
+    for key in range(len(stat_dict)):
+        bar_chart.add('{0}-{1}'.format(lengend, key), stat_dict.get(key))
```

Comment:
This works well if `stat_dict` has keys that are integers, and are in the range 0-N:

```python
>>> d = {0: 10, 1: 10, 2: 10}
>>> d.keys()
dict_keys([0, 1, 2])
>>> range(len(d))
range(0, 3)
>>> list(d.keys()) == list(range(len(d)))  # note the type casts
True
```

But this code does not work well if unexpected keys are present.

```python
>>> d = {1: 2, 3: 4}
>>> d.keys()
dict_keys([1, 3])
>>> range(len(d))
range(0, 2)
>>> list(d.keys()) == list(range(len(d)))
False
```

The problem with the approach used is that, if unexpected keys are present, the user will have a hard time troubleshooting the problem. No dict lookup errors are ever raised. It should be possible to do this:

```python
keys = list(stat_dict.keys())
keys.sort()
for key in keys:
    bar_chart.add('{0}-{1}'.format(legend, key), stat_dict[key])
```

And whatever code calls `generate_bar_chart_stat` should worry about passing in sane data.

There's no need to fix this code right now if it cannot easily be fixed. But consider a future refactor to address this issue.